### PR TITLE
feat: optimize lodash for esm/mjs + add bundleDeps option

### PR DESCRIPTION
**Pull request by marbemac**
- Always optimize lodash imports for compat with esm/mjs.
- Custom build of https://github.com/stoplightio/rollup-plugin-optimize-lodash-imports to add `.js` to end of imports, can replace if it is accepted upstream (will create a PR). https://github.com/stoplightio/rollup-plugin-optimize-lodash-imports/tree/fix/js-extension-for-mjs-swc
- Added ability to force a specific dep to be bundled via the `bundleDeps` config in package.json

Yalc'd and tested over in the elements repository.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "/bin",
     "/dist",
     "/templates",
+    "/rollup-plugin-optimize-lodash-imports",
```
Comment (P0lip) [inline package.json:21]: feels like a good opportunity for placing it under the old-fashioned `vendor` directory.
What do you think? This way, we can place more packages if needed, and one doesn't need to remember about adding them to `package.json`

Comment (marbemac) [reply package.json:21]: is vendor included by default or something?

File: README.md [hunk 2]
```diff
@@ -187,3 +187,17 @@ Simply add a `release` property to your `package.json` file. For example:
...
+
+## Rollup
+
+By default all `dependencies` and `peerDependencies` declared in your `package.json` will be treated as external deps during `sl-scripts bundle`. If you would like to always bundle a dep, list those deps in your `package.json` file like so:
+
+```json
+{
+  "name": "your-package",
+  "version": "0.0.0",
+  "rollup": {
```
Comment (P0lip) [inline README.md:199]: I hope Rollup never decides to support some configuration via package.json 🙈

File: rollup.config.js [hunk 3]
```diff
@@ -31,7 +37,15 @@ const dependencies = [
   ...Object.keys(packageJson.dependencies || {}),
   ...Object.keys(packageJson.peerDependencies || {}),
 ];
-const external = module => dependencies.some(dep => module === dep || module.startsWith(`${dep}/`));
+const rollupConfig = packageJson.rollup || {};
+const alwaysInclude = rollupConfig.bundleDeps || [];
+
+const external = module => {
+  const isDefaultExternal =  dependencies.some(dep => module === dep || module.startsWith(`${dep}/`));
```
Comment (P0lip) [inline rollup.config.js:44]:
```suggestion
  const isDefaultExternal = dependencies.some(dep => module === dep || module.startsWith(`${dep}/`));
```

**Review by P0lip — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (marbemac) [reply README.md:199]: lol yeah - i mean i could pick a different key? what do you think, should we?

Comment (P0lip) [reply README.md:199]: `sl_rollup`? I don't really know, something with Stoplight would be probably best.
Or maybe `@stoplight/scripts`.

Comment (marbemac) [reply README.md:199]: Meh, we'll be using this very very rarely - let's just go with it for now, can always update down the road in the (pretty unlikely) event that there's a conflict in future.

Comment (P0lip) [reply README.md:199]: Yeah, fair enough.

**Comment by marbemac** [comment]
Ugh, these two "build elements" tests have been failing for months :(

Screen Shot 2021-11-23 at 7 58 07 PM

**Comment by marbemac** [comment]
Not fixing this separate issue as part of this PR, going to merge.

**Pull request merged**